logo.js

```diff
@@ -546,6 +552,32 @@ function LogoInterpreter(turtle, stream, savehook)
     return lhs;
   }
 
+  function defer(func) {
+    var input = Array.prototype.slice.call(arguments, 1);
+    return function () {
```

Comment:
Ah. :) Then let's factor out a helper function like Promise.all() but enforces the order.